### ISSUE TITLE
Overwrite metadata attributes with on-chain attributes

### ISSUE THREAD

**Issue by vikiival** [kodadot/nft-gallery#6785]
I don't know what you think but imo the attributes from `set_attribute` extrinsics should overwrite those from the metadata in case of duplicates. aka

```js
//pseudocode
const nftProperties = {
  ...attributesFromMetadata,
  ...attributesFromExtrinsics,
}
```

_Originally posted by @niklasp in https://github.com/kodadot/nft-gallery/issues/6774#issuecomment-1683860996_

**Comment by niklasp** [comment]
That means in practice its currently the other way around? E.g. when a NFT has two images defined, one in metadata and one in on chain attributes, the "first" one is picked  from the metadata?

Also for other random attributes with the same name, both will appear in the Properties tab of the UI?

**Comment by vikiival** [comment]
> That means in practice its currently the other way around? E.g. when a NFT has two images defined, one in metadata and one in on chain attributes, the "first" one is picked from the metadata?


current solutuion is based on the **concat strategy**

[🩵, 💚] + [🩵, 💙] = [🩵, 💚, 🩵, 💙]

**Comment by stephenjason89** [comment]
@vikiival hello. Is this issue taken?

**Comment by niklasp** [comment]
yeah i know concat but what does the ui display in above case when e.g. 🩵 is the NFT description

**Comment by vikiival** [comment]
> display in above case when e.g. 🩵 is the NFT description

Currently it is stored under attributes. 


_I see your point now_

If you want to override nft metadata we can either proceed with KodaDot conditional package (if-that). Or please open an issue  in stick repo how it should work. 

**How we handle metadata**

Currently metadata ipfs string is a foreign key to the _MetadataEntity_ which is loaded from IPFS.
Mutating the ME directly would result into undefined behaviour as ME could be shared with many other  NFTs. 
What my brain says rn:

**Comment by niklasp** [comment]
@vikiival i tested the following setup

1. created a NFT on asset hub with `nft.mint`
1. then called `nft.setMetadata` on that NFT (https://kodadot.xyz/ahk/gallery/101-88888888)
1. the metadata attributes all showed in the "Properties Tab"
1. Then i did a `nft.setAttribute` on that nft with `key: something` `value:223`
1. now only that single attribute shows in the "Properties" Tab

=> metadata attributes are not merged but overriden completely by the `attributes` from `setAttribute` extrinsics, as far as i can tell

**Comment by roiLeo** [comment]
- link #8425
